File: 0739-daily-temperatures/0739-daily-temperatures.ts
function dailyTemperatures(temperatures: number[]): number[] {
    let stack = []
    const ans = new Array(temperatures.length).fill(0);

    for (let i = 0; i < temperatures.length; i++) {
        while (temperatures[i] > temperatures[stack[stack.length-1]] && stack.length) {
            const j = stack.pop()
            ans[j] = i - j
        }
        stack.push(i)
    }
    return ans
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        ans = [0] * len(temperatures)
        stack = []

        for i in range(len(temperatures)):
            while len(stack) > 0 and temperatures[stack[-1]] < temperatures[i]:
                idx = stack.pop()
                ans[idx] = i - idx

            stack.append(i)
        return ans